test/pylib/scylla_cluster.py

```diff
@@ -838,6 +852,24 @@ def setLogger(self, logger: logging.LoggerAdapter):
         for srv in self.servers.values():
             srv.setLogger(self.logger)
 
+    async def change_ip(self, server_id: ServerNum) -> ActionReturn:
+        """Lease a new IP address and update conf/scylla.yaml with it. The
+        original IP is released at the end of the test to avoid an
+        immediate recycle within the same cluster."""
+        if server_id not in self.servers:
+            return ScyllaCluster.ActionReturn(success=False, msg=f"Server {server_id} unknown")
+        server = self.servers[server_id]
+        if server.is_running:
+            msg = f"Server {server_id} is running: stop it first and then change its ip"
+            return ScyllaCluster.ActionReturn(success=False, msg=msg)
```

Comment:
It's awkward that we turn this into error code `404` (in `_server_change_ip`).

Since the server must be stopped - it is a precondition, and not satisfying it would be a bug in the test itself - I think we can just `assert` inside `ScyllaServer.change_ip` and don't have to check it here. If the assertion fires, it will be turned into error code `500`.

(We could say the same for the check above... but that's a larger refactor)

Anyway, the precondition that the server identified by `server_id` must be stopped should also be stated in the docstring for `ScyllaCluster.change_ip`.

Reply:
HTTP 500 is harder to debug. It would be nice to assert on manager_client, but it doesn't contain the server state right now which is sad.

Reply:
I added the comment.